### ISSUE TITLE
How long does it take to train?

### ISSUE THREAD

**Issue by peng666** [paul007pl/VRCNet#3]
How long does it take to train? I train PCN about 3 epochs a day.

**Comment by paul007pl** [comment]
I remember that it should take around a few hours for training pcn (2048pts, 100 epochs, with CD loss).
I would like to suggest that you can discard the emd loss evaluation (very slow, especially for high resolution), which can help you save time.
Good Luck~

**Comment by peng666** [comment]
But I've trained with CD as a loss function, still three generations a day. I'm using a dense point cloud (16384 points). Is there any mistake here?

**Comment by paul007pl** [comment]
I know you are using CD for training, but please check if you still evaluate the emd loss during training.
It is reasonable that you should take a longer time for training with 16384 points.
If I did not remember wrongly, it takes around 1 day for training PCN (16384 points, CD loss, 100 epochs) with a V100 GPU.
I suggest that you can try 2048 first. (I also did in this way...)
If you method works better than the others on 2048 pts, it should probably outperform the others on 16384 pts also.

**Comment by peng666** [comment]
Your suggestion is very good. Thank you very much！
